controllers/sandbox_controller.go

```diff
@@ -76,22 +79,66 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	oldStatus := sandbox.Status.DeepCopy()
+
 	// Create a hash from the sandbox.Name and use it as label value
 	nameHash := NameHash(sandbox.Name)
-	_, err := r.reconcilePod(ctx, sandbox, nameHash)
-	if err != nil {
-		return ctrl.Result{}, err
+	var conditions []*metav1.Condition
+	condition, err := r.reconcilePod(ctx, sandbox, nameHash)
+	conditions = append(conditions, condition)
+	if err == nil {
+		condition, err = r.reconcileService(ctx, sandbox, nameHash)
+		conditions = append(conditions, condition)
 	}
+	updateErr := r.updateStatus(ctx, oldStatus, sandbox, conditions)
+	return ctrl.Result{}, errors.Join(err, updateErr)
+}
 
-	_, err = r.reconcileService(ctx, sandbox, nameHash)
-	if err != nil {
-		return ctrl.Result{}, err
+func (r *SandboxReconciler) updateStatus(ctx context.Context, oldStatus *sandboxv1alpha1.SandboxStatus, sandbox *sandboxv1alpha1.Sandbox, conditions []*metav1.Condition) error {
+	log := log.FromContext(ctx)
+	readyCondition := metav1.Condition{
+		Type:               string(sandboxv1alpha1.SandboxConditionReady),
+		Status:             metav1.ConditionTrue,
+		Reason:             "DependenciesReady",
+		Message:            "",
+		ObservedGeneration: sandbox.Generation,
+	}
+	conditionsPresent := false
+	for _, condition := range conditions {
+		if condition == nil {
+			continue
+		}
+		conditionsPresent = true
```

Comment:
A couple of questions:
1. Is it possible when there is only one true condition available (others are nil)? Do we still consider it "DependenciesReady"?
2. Is it possible when there is only one false condition available (others are nil)? If so, will "DependenciesNotReady" cause some confusion as there is only one dependency has clear message?

Reply:
re:
1. no we should not. fixing it.
2. if some are nil  => err would be set and it would re-reconcile soon. so temporary.